rust_dev_preview/examples/auto-scaling/src/bin/scenario.rs

```diff
@@ -0,0 +1,170 @@
+    let stable = scenario.wait_for_stable(2).await;
+    if let Err(err) = stable {
+        warnings.push(
+            "There was a problem while waiting for group to be stable",
+            err,
+        );
+    }
+
+    // 8. DescribeAutoScalingInstances: show that two instances are launched.
+    show_scenario_description(
+        &scenario,
+        "show that two instances are launched after setting desired capacity",
+    )
+    .await;
+
+    let ids_before = scenario
+        .list_instances()
+        .await
+        .map(|v| v.into_iter().collect::<BTreeSet<_>>())
+        .unwrap_or_default();
+
+    // 9. TerminateInstanceInAutoScalingGroup: terminate one of the instances in the group.
+    let terminate_some_instance = scenario.terminate_some_instance().await;
+    if let Err(err) = terminate_some_instance {
+        warnings.push("There was a problem replacing an instance", err);
+    }
+
+    let wait_after_terminate = scenario.wait_for_stable(1).await;
+    if let Err(err) = wait_after_terminate {
+        warnings.push(
+            "There was a problem waiting after terminating an instance",
+            err,
+        );
+    }
+
+    let wait_scale_up_after_terminate = scenario.wait_for_stable(2).await;
+    if let Err(err) = wait_scale_up_after_terminate {
+        warnings.push(
+            "There was a problem waiting for scale up after terminating an instance",
+            err,
+        );
+    }
+
+    let ids_after = scenario
+        .list_instances()
+        .await
+        .map(|v| v.into_iter().collect::<BTreeSet<_>>())
+        .unwrap_or_default();
+
+    let difference = ids_after.intersection(&ids_before).count();
+    if !(difference == 1 && ids_before.len() == 2 && ids_after.len() == 2) {
+        warnings.push(
+            "Before and after set not different",
+            anyhow!("{difference}"),
+        );
+    }
+
+    // 10. DescribeScalingActivities: list the scaling activities that have occurred for the group so far.
+    show_scenario_description(
+        &scenario,
+        "list the scaling activities that have occurred for the group so far",
+    )
+    .await;
+
+    // 11. DisableMetricsCollection
+    let scale_group = scenario.scale_group_to_zero().await;
+    if let Err(err) = scale_group {
+        warnings.push("There was a problem scaling the group to 0", err);
+    }
+    show_scenario_description(&scenario, "Scenario scaled to 0").await;
+
+    // 12. DeleteAutoScalingGroup (to delete the group you must stop all instances):
+    // 14. Delete LaunchTemplate.
+    let clean_scenario = scenario.clean_scenario().await;
+    if let Err(err) = clean_scenario {
+        warnings.push("There was a problem cleaning the scenario", err);
+    } else {
+        info!("The scenario has been cleaned up!");
+    }
+
+    if warnings.is_empty() {
+        Ok(())
+    } else {
+        Err(anyhow!(
+            "There were warnings during scenario execution:\n{warnings:?}"
```

Comment:
you're not going to get a particularly nice output debug logging a struct like this

Reply:
Agreed. Replaced with ScenarioError struct.